src/main/java/com/example/inventoryservice/service/impl/IngredientServiceImpl.java

```diff
@@ -0,0 +1,93 @@
+        UserDto userDto = userService.getCurrentUser();
+        RestaurantDto restaurantDto = userDto.getRestaurant();
+        ingredientDto.setRestaurant(restaurantDto);
+        Ingredient ingredient = ingredientConverter.dtoToEntity(ingredientDto);
+        if (ingredientExists(ingredient)) {
+            throw new ServiceException("Not unique ingredient");
+        }
+        Ingredient persistIngredient = ingredientRepository.save(ingredient);
+        return ingredientConverter.entityToDto(persistIngredient);
+    }
+
+    @Override
+    public IngredientDto updateAmount(IngredientDto ingredientDto) {
+        logger.info("Update ingredient amount");
+        Ingredient ingredient = ingredientConverter.dtoToEntity(ingredientDto);
+        if (!ingredientExists(ingredient)) {
+            logger.error("No such ingredient {}", ingredient.getName());
+            throw new ServiceException("No such ingredient");
+        }
+        Ingredient persistIngredient = ingredientRepository.findById(ingredient.getId())
+                                                           .orElse(new Ingredient());
+        BigDecimal persistAmount = persistIngredient.getAmount();
+        persistIngredient.setAmount(persistAmount.add(ingredient.getAmount()));
+        persistIngredient = ingredientRepository.save(persistIngredient);
+        return ingredientConverter.entityToDto(persistIngredient);
+    }
+
+    @Override
+    public IngredientDto updatePrice(IngredientDto ingredientDto) {
+        logger.info("Update ingredient price");
+        Ingredient ingredient = ingredientConverter.dtoToEntity(ingredientDto);
+        if (!ingredientExists(ingredient)) {
+            logger.error("No such ingredient {}", ingredient.getName());
+            throw new ServiceException("No such ingredient");
+        }
+        Ingredient persistIngredient = ingredientRepository.findById(ingredient.getId())
+                                                           .orElse(new Ingredient());
+        BigDecimal persistPrice = persistIngredient.getPrice();
+        persistIngredient.setPrice(persistPrice.add(ingredient.getAmount()));
+        persistIngredient = ingredientRepository.save(persistIngredient);
+        return ingredientConverter.entityToDto(persistIngredient);
+    }
+
+    @Override
+    public List<IngredientDto> findAllByRestaurant() {
+        logger.info("Find all ingredients by restaurant");
+        UserDto userDto = userService.getCurrentUser();
+        List<Ingredient> ingredients = ingredientRepository.findAllByRestaurant_Id(userDto.getRestaurant()
+                                                                                          .getId());
+        return ingredientConverter.entityToDto(ingredients);
+    }
+
+    @Override
+    public boolean ingredientExists(Ingredient ingredient) {
+        return ingredientRepository.findAllByNameIgnoreCaseAndRestaurant_Id(ingredient.getName(), ingredient.getRestaurant()
```

Comment:
[medium] Checking if something exists in the DB is better done using the `count` query or `select true where ...` rather than querying actual data which can be huge.